tests/e2e/util.go

```diff
@@ -805,6 +809,119 @@ func invokeVCenterServiceControl(command, service, host string) error {
+	createGCURL := "https://" + wcpHost + tkgAPI
+	framework.Logf("URL %v", createGCURL)
+	tkg_yaml, err := filepath.Abs(gcManifest + "tkg.yaml")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	framework.Logf("Takinig yaml from %v", tkg_yaml)
+	gcBytes, err := ioutil.ReadFile(tkg_yaml)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	req, err := http.NewRequest("POST", createGCURL, bytes.NewBuffer(gcBytes))
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	req.Header.Add("Authorization", "Bearer "+wcpToken)
+	req.Header.Add("Accept", "application/yaml")
+	req.Header.Add("Content-Type", "application/yaml")
+
+	resp, err := client.Do(req)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	defer resp.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	framework.Logf("API Response status %d", resp.StatusCode)
+	gomega.Expect(resp.StatusCode).Should(gomega.BeNumerically("==", 201))
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	response := string(bodyBytes)
+	framework.Logf(response)
+
+}
+
+//getGC polls for the GC status, returns error if its not in running phase
+func getGC(wcpHost string, wcpToken string, gcName string) error {
+	transCfg := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // ignore expired SSL certificates
+	}
+
+	client := &http.Client{Transport: transCfg}
+	getGCURL := "https://" + wcpHost + tkgAPI + gcName
+	framework.Logf("URL %v", getGCURL)
+
+	req, err := http.NewRequest("GET", getGCURL, nil)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	req.Header.Add("Authorization", "Bearer "+wcpToken)
+
+	waitErr := wait.Poll(pollTimeout, pollTimeout*6, func() (bool, error) {
+		framework.Logf("Polling for New GC status")
+		resp, err := client.Do(req)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer resp.Body.Close()
+
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		framework.Logf("API Response status %d", resp.StatusCode)
+		gomega.Expect(resp.StatusCode).Should(gomega.BeNumerically("==", 200))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		response := string(bodyBytes)
+
+		if strings.Contains(response, "\"phase\":\"running\"") {
+			framework.Logf("new gc is up and running")
+			framework.Logf(response)
+			return true, nil
+		}
+		return false, nil
+	})
+
+	return waitErr
+
+}
+
+//getWCPSessionId returns the bearer token for given user
+func getWCPSessionId(hostname string, username string, password string) string {
+	type WcpSessionID struct {
+		//Session_id string `json: "session_id"`
+		Session_id string
+	}
+
+	transCfg := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // ignore expired SSL certificates
+	}
+
+	client := &http.Client{Transport: transCfg}
+	URL := "https://" + hostname + "/wcp/login"
+	framework.Logf("URL %s", URL)
+
+	req, err := http.NewRequest("POST", URL, nil)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	req.Header.Add("Accept", "application/json")
```

Comment:
Have a method to do https(s) get and post calls, for those methods have a param to indicate whether to use json or yaml content and other param for any additional headers required and/or auth. Fetch response and do response code validation and return the response. Instead of having this logic in multiple places.

Reply:
I have added two methods `httpGet` and `httpPost` to perform GET and POST calls, with slight modifications to your comment. `httpPost` will not do response code validation but it will return response code Because some APIs will return 200 and 201